### PR TITLE
 issue-1350 - set correct AI for mini pet 

### PR DESCRIPTION
**Pull request by RichardGe2**
here is my suggestion to fix 
https://github.com/cmangos/issues/issues/1350

### CODE REVIEW

**Comment by killerwife** [comment]
a better solution would be to enforce REACT_PASSIVE for mini pets. You can do that for any AI now, so if you want to fix this nicely, on summoning mini pet set REACT_PASSIVE. I believe there is an entry point for it somewhere.

**Comment by RichardGe2** [comment]
ok, I'll modify my pull request later, in the coming days..

**Comment by RichardGe2** [comment]
@killerwife ,
But wait, we already have  
case MINI_PET:
SetReactState(REACT_PASSIVE);
inside PetAI.cpp
the problem I'm fixing here is that mini pet was assigned to AI  "GuardianAI" instead of "PetAI"

**Comment by killerwife** [comment]
Then we need to set it at minipet creation, not inside AIs so that we prevent this issue forever into the future.

**Comment by killerwife** [comment]
bool Spell::DoSummonCritter(CreatureSummonPos
This is the func where we create all minipets. If you set REACT_PASSIVE here, it should work.

**Comment by RichardGe** [comment]
ok, thanks, I'll update my pull request later this week

**Comment by RichardGe2** [comment]
ok thanks, I'll update my pull request later this week

**Comment by RichardGe2** [comment]
@killerwife , let me know if you are okay now :)

File: src/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -5054,6 +5054,11 @@ void Spell::EffectSummonCritter(SpellEffectIndex eff_idx)
         m_originalCaster->AI()->JustSummoned(critter);
     else if (m_caster->AI())
         m_caster->AI()->JustSummoned(critter);
+
+    //for MINI_PET we ensure to set a passive react
+    CreatureAI* critterAI = critter->AI();
+    if ( critterAI ) { critterAI->SetReactState(REACT_PASSIVE); }
+
```
Comment (killerwife) [inline src/game/Spells/SpellEffects.cpp]: cmangos codestyle:
if (CreatureAI* critterAI = critter->AI())
    critterAI->SetReactState(REACT_PASSIVE);

**Comment by killerwife** [comment]
Sorry for being annoying yet again. Do 2 more things and I will merge it. Move the code above the JustSummoned. Then remove the previous commits (so that only the final one remains). You can force push into a PR branch.

**Comment by RichardGe2** [comment]
no problem, Just to clarify, if I remove the first commit, the mini pet IA will be changed from "PetAI" to "GuardianAI". Do we want that ?

**Comment by killerwife** [comment]
It doesn't matter. Out of each AI, the pets only utilize follow movement. Out of Guardian AI you can use some EAI events to actually scripted, PetAI can't do any of that. So its actually for the best.

**Comment by killerwife** [comment]
Squash the whole PR into one commit.

**Comment by RichardGe2** [comment]
Done.  I double checked on my server, it seems to work fine

**Comment by RichardGe2** [comment]
I think I have done a bad action, - not comfortable with git rebase -  I'll fix that later

**Comment by RichardGe2** [comment]
ok I think my pull request is good now, 
( I don't think I can squash the  "Merge pull request #1" and "Merge pull request #2"  right ? )

**Comment by cala** [comment]
Before I merge this, do you know if it also applies to TBC and WotLK ?

**Comment by killerwife** [comment]
@cala it applies to all branches. The PetAI setting needs to be removed (with MINI_PET). This is a much better solution than I did when adding ReactStates to all AIs.

**Comment by cala** [comment]
Closed by https://github.com/cmangos/mangos-classic/commit/49252e28ee851b9171ef4a2d58699876ec599de5

**Pull request closed without merging**